diff -r -u -P include.orig/stdio.h include/stdio.h
--- include.orig/stdio.h	2004-11-28 10:07:16.000000000 +0100
+++ include/stdio.h	2004-11-28 10:09:20.000000000 +0100
@@ -579,8 +579,11 @@
 
 
 #ifdef	__USE_POSIX
+#	ifndef ___ctermid_defined
+#	define ___ctermid_defined
 /* Return the name of the controlling terminal.  */
 extern char *ctermid (char *__s) __THROW;
+#	endif
 #endif /* Use POSIX.  */
 
 
diff -r -u -P include.orig/unistd.h include/unistd.h
--- include.orig/unistd.h	2004-11-28 10:01:39.000000000 +0100
+++ include/unistd.h	2004-11-28 10:08:51.000000000 +0100
@@ -963,7 +963,10 @@
    It is also found in <stdio.h>.  */
 #ifdef __USE_XOPEN
 /* Return the name of the controlling terminal.  */
+#	ifndef ___ctermid_defined
+#	define ___ctermid_defined
 extern char *ctermid (char *__s) __THROW;
+#	endif
 #endif
 
 
